pennylane/qinfo/transforms.py

```diff
@@ -42,42 +44,65 @@ def reduced_dm(qnode, wires):
         import numpy as np
 
         dev = qml.device("default.qubit", wires=2)
+
         @qml.qnode(dev)
         def circuit(x):
-          qml.IsingXX(x, wires=[0,1])
-          return qml.state()
+            qml.IsingXX(x, wires=[0,1])
+            return qml.state()
 
-    >>> reduced_dm(circuit, wires=[0])(np.pi/2)
-     [[0.5+0.j 0.+0.j]
-      [0.+0.j 0.5+0.j]]
+    >>> transformed_circuit = reduced_dm(circuit, wires=[0])
+    >>> transformed_circuit(np.pi/2)
+    tensor([[0.5+0.j, 0. +0.j],
+            [0. +0.j, 0.5+0.j]], requires_grad=True)
 
     .. seealso:: :func:`pennylane.density_matrix` and :func:`pennylane.math.reduce_dm`
     """
-    wire_map = {w: i for i, w in enumerate(qnode.device.wires)}
+    all_wires = kwargs.get("device_wires", tape.wires)
+    wire_map = {w: i for i, w in enumerate(all_wires)}
     indices = [wire_map[w] for w in wires]
 
-    def wrapper(*args, **kwargs):
-        qnode.construct(args, kwargs)
-        measurements = qnode.tape.measurements
-        if len(measurements) != 1 or not isinstance(measurements[0], StateMP):
-            raise ValueError("The qfunc measurement needs to be State.")
+    measurements = tape.measurements
+    if len(measurements) != 1 or not isinstance(measurements[0], StateMP):
+        raise ValueError("The qfunc measurement needs to be State.")
+
+    def processing_fn(res):
+        device = kwargs.get("device", None)
```

Comment:
Can you add a small comment that is is possible thanks to the custom qnode transform?